ParcialProgramacion4/src/Empleado.java

```diff
@@ -60,4 +60,57 @@ public void mostrarHabilidades() {
         }
     }
 
+
+    public boolean puedeAplicar(Hashtable<Habilidad,Integer> requisitos) {
+        //si esta en un puesto jerarquico, determino si cumple con annos minimos en ese puesto
+        //comparo requisitos (lo q pide puesto) con habilidades (lo q tiene el empleado) 
+        Cargo cargoActual = this.getCargoActual();
+        
+        return cargoActual.jerarquicoCumpleAnnosMinimos() && this.cumpleRequisitos(requisitos);
+    }
+
+    public Cargo getCargoActual() {
+        return this.historialDeCargos.get(historialDeCargos.size() - 1);
+    }
+
+    public boolean cumpleRequisitos(Hashtable<Habilidad,Integer> requisitos) {
+        //recorro la hashtable de requisitos y me fijo si tiene la habilidad el empleado y los años minimos
+        boolean cumpleReq = true; //si no tiene algun requisito o años necesarios, no sigue recorriendo
+        Habilidad requisito;
+
+        Enumeration<Habilidad> enumReq = requisitos.keys();
+        while (enumReq.hasMoreElements() && cumpleReq) {
+            requisito = enumReq.nextElement();
+
+            if (habilidades.containsKey(requisito)) {
+                            //   años de la persona    >= años necesarios convocatoria
+                cumpleReq = habilidades.get(requisito) >= requisitos.get(requisito); 
+
+            } else {
+                cumpleReq = false; //no tiene ese requisito, ya no puede aplicar
+            }
+        }
+
+        return cumpleReq;
+    }
+
+
+    public int getAnnosEnEmpresa() {
+        //calcular años y luego determino si tengo que restarle porque no llego mes o dia para pasar de año
+        Fecha hoy = Fecha.hoy();
+        int annosEnEmpresa = hoy.getAño() - fechaDeIngreso.getAño();
+ 
+        //Determino si tengo que restarle un año de mas
+        if (hoy.getMes() < fechaDeIngreso.getMes()) {
+             annosEnEmpresa--; //todavia no paso el mes de ingreso para contabilizar el año ej fecha hoy 2023/11/29 y fecha inicio 2022/12/20
+        } else {
+             if (hoy.getMes() == fechaDeIngreso.getMes() && hoy.getDia() < fechaDeIngreso.getDia()) {
+                 annosEnEmpresa--; //todaia no paso el dia del ingreso en el mismo mes para contabilizar el año ej fecha hoy 2023/11/29 y fecha inicio 2022/11/30
+             }
+        }
```

Comment:
```suggestion
        if (fechaDeIngreso.compareTo(hoy) < 0) {
            annosEnEmpresa--;
        }
```
Creo que se podria hacer asi